flintrock/ec2.py

```diff
@@ -274,20 +289,19 @@ def check_network_config(*, region_name: str, vpc_id: str, subnet_id: str):
             "See: https://github.com/nchammas/flintrock/issues/43"
             .format(v=vpc_id)
         )
-    if not ec2.Subnet(subnet_id).map_public_ip_on_launch:
-        raise ConfigurationNotSupported(
-            "{s} does not auto-assign public IP addresses. "
-            "Flintrock requires public IP addresses.\n"
-            "See: https://github.com/nchammas/flintrock/issues/14"
-            .format(s=subnet_id)
-        )
+    use_private_vpc = not ec2.Subnet(subnet_id).map_public_ip_on_launch
```

Comment:
As noted earlier, I think this belongs as a property of `EC2Cluster`. I think all we need to do here is remove the `ConfigurationNotSupported` for private VPCs, as you've done.